x/intertx/module/module.go

```diff
@@ -0,0 +1,100 @@
+package module
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/tendermint/tendermint/abci/types"
+
+	"github.com/regen-network/regen-ledger/x/intertx"
+	intertxClient "github.com/regen-network/regen-ledger/x/intertx/client"
+	"github.com/regen-network/regen-ledger/x/intertx/keeper"
+	v1 "github.com/regen-network/regen-ledger/x/intertx/types/v1"
+)
+
+var (
+	_ module.AppModule = AppModule{}
+)
+
+// AppModule implements the AppModule interface for the capability module.
+type AppModule struct {
+	keeper keeper.Keeper
+}
+
+func NewModule(k keeper.Keeper) AppModule {
+	return AppModule{k}
+}
+
+func (a AppModule) Name() string {
+	return intertx.ModuleName
+}
+
+func (a AppModule) RegisterLegacyAminoCodec(amino *codec.LegacyAmino) {
+	v1.RegisterCodec(amino)
+}
+
+func (a AppModule) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
+	v1.RegisterInterfaces(registry)
+}
+
+func (a AppModule) DefaultGenesis(_ codec.JSONCodec) json.RawMessage {
+	return nil
+}
+
+func (a AppModule) ValidateGenesis(_ codec.JSONCodec, _ client.TxEncodingConfig, _ json.RawMessage) error {
+	return nil // TODO(Tyler): validate here?
```

Comment:
Are you planning to add genesis state in this PR?

Reply:
removed the TODO, i checked ica-demo-app and gravity bridge's implementation and they do not do any genesis validation 🤷🏻

Reply:
Ah right. We don't have any state data in `intertx` module.